packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts

```diff
@@ -275,9 +275,9 @@ export default ({ api, coreSagas, networks }) => {
         const countryCode = country || 'US'
         const currency = guessCurrencyBasedOnCountry(countryCode)
 
+        yield put(actions.modules.settings.updateCurrency(currency, true))
         yield put(actions.core.settings.setCurrency(currency))
-        // fetch settings again
-        yield call(coreSagas.settings.fetchSettings)
+
```

Comment:
Doesn't updateCurrency call setCurrency?

Reply:
It does, but without this like I run into NaN issue
![image](https://user-images.githubusercontent.com/67264242/134017702-e0ee7d21-1682-4c4b-ba19-7dacdc65ccda.png)

Reply:
Why do you need to call `yield put(actions.core.settings.setCurrency(currency))`

Reply:
Let me try removing it, if all works as expected we can remove it.

Reply:
great call, removed.